src/MIDebugEngine/Engine.Impl/Breakpoints.cs

```diff
@@ -96,7 +96,20 @@ private static MIBreakpointState StringToBreakpointState(string addr)
             process.VerifyNotDebuggingCoreDump();
 
             string basename = System.IO.Path.GetFileName(documentName);     // get basename from Windows path
-            return EvalBindResult(await process.MICommandFactory.BreakInsert(process.EscapePath(basename), line, condition, ResultClass.None), pbreak);
+            basename = process.EscapePath(basename);
+            BindResult bindResults = EvalBindResult(await process.MICommandFactory.BreakInsert(basename, line, condition, ResultClass.None), pbreak);
+
+            // On GDB, the returned line information is from the pending breakpoint instead of the bound breakpoint.
+            // Check the address mapping to make sure the line info is correct.
+            if (process.MICommandFactory.Mode == MIMode.Gdb)
+            {
+                foreach (var boundBreakpoint in bindResults.BoundBreakpoints)
```

Comment:
I think this collection will be null if the error message constructor is used (see line 58)